src/main/java/core/basesyntax/service/impl/FruitListImpl.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.db.FruitTransaction;
+import core.basesyntax.service.FruitList;
+import java.util.ArrayList;
+import java.util.List;
+
+public class FruitListImpl implements FruitList {
+    @Override
+    public List<FruitTransaction> getFruitList(List<FruitTransaction> list) {
+        List<String> listOfFruitsString = list.stream()
+                .map(FruitTransaction::getFruit)
+                .distinct()
+                .toList();
+        List<FruitTransaction> listOfFruitsAsObjects = new ArrayList<>();
```

Comment:
Instead of creating a `List<String>` and then converting it to a `List<FruitTransaction>`, consider streamlining the process by directly creating a `List<FruitTransaction>` with distinct fruits. This will make the code more efficient and easier to understand.